src/librustc_resolve/lifetimes.rs

```diff
@@ -1317,7 +1331,8 @@ impl<'a, 'tcx> LifetimeContext<'a, 'tcx> {
             is_in_fn_syntax: self.is_in_fn_syntax,
             labels_in_fn,
             xcrate_object_lifetime_defaults,
-            lifetime_uses: lifetime_uses,
+            lifetime_uses,
+            missing_named_lifetime_spots: missing_named_lifetime_spots.to_vec(),
```

Comment:
```suggestion
            missing_named_lifetime_spots: missing_named_lifetime_spots.clone(),
```
Otherwise it looks like these are two different fields with different types.

Reply:
Also, this clone happens on a happy path and not only during error reporting.
Can it use `mem::take` instead like `labels_in_fn` and `xcrate_object_lifetime_defaults` above?

Reply:
done